WordPress/openverse
# Cleanup tag display for long lists of tags

## Fixes



Fixes #2589 by @zackkrida 

## Description


This PR hides the tags beyond the third row and shows a "Show more tags" button if there are hidden tags.

I first tried simply hiding the tags setting the max height and using `overflow-hidden`. However, this way,  you can navigate to the hidden tags using <kbd>Tab</kbd>. Even though those tags are not visible, they are focusable and are read by the screenreader.

That's why I added a function that detects the index of the tag that is the first on the 4th line of tags, and set the `visibleTags` value to the tags up to the "first on the 4th line". When the user clicks show more, the `visibleTags` is set to all of the tags, and when they click on hide tags, the `visibleTags` array only contains the first 3 lines.

For counting the rows of tags, I used `offsetLeft`, but had to make comparison operator (`<` or `>`) conditional to make it work with RTL (some of the first VR tests here for RTL are failures because I only used one comparison)

The expander button has a `aria-expanded` set to `true` or `false` based on whether the tags are all shown, or hidden.

## Testing Instructions

Enable the `additional_search_views` on the `/preferences` page (note that this is only implemented for when this flag is on)
Go to an image result with many tags, such as `[http://localhost:8443/image/7e138814-e774-45fb-91ee-7b728c174616?q=Titmouse](http://localhost:8443/image/7e138814-e774-45fb-91ee-7b728c174616?q=Titmouse)`. 
Check that only the 3 rows of tags are displayed, with a "Show more tags" button. Try clicking on the button. The first opened tag should be focused, and the "See less" button should be displayed.
Check that this works for pages with other locales, and that the tags for media with fewer tags are displayed as before.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
